pkg/mapper/upsert.go

```diff
@@ -177,20 +177,26 @@ func (b *AuthMapper) upsertAuth(args *MapperArguments) error {
 	return UpdateAuthMap(b.KubernetesClient, authData, configMap)
 }
 
-func upsertRole(authMaps []*RolesAuthMap, resource *RolesAuthMap) ([]*RolesAuthMap, bool) {
+func upsertRole(authMaps []*RolesAuthMap, resource *RolesAuthMap, appendGroups, dontUpdateUnsername bool) ([]*RolesAuthMap, bool) {
```

Comment:
It would probably be more scalable for future changes to add a struct `UpsertOptions` that can contain these two variables.

Reply:
on it

Reply:
you want a method to set the options to and return that struct or just do it inline?